torch/csrc/jit/codegen/cuda/compute_at_map.cpp

```diff
@@ -1136,8 +1178,32 @@ void ComputeAtMap::buildUniqueExactExprMaps() {
        id_graph_.exactNodes().disjointSets()) {
     std::vector<Expr*> definitions;
 
+    // N^2 in number of unique transformations, this might be better to do
+    // when generating the map.
     for (auto id : disjoint_set_shared_ptr->vector()) {
       if (id->definition() != nullptr) {
+        auto id_inputs =
```

Comment:
I guess having this type of exprs in the unique def set cause a loop when traversing the ID graph?

Reply:
Didn't end up stuck in a loop in the traversal. Traversal only visited the set once since we only allow traversing a group once. So it didn't try to re-traverse that group but also didn't mark the group as an input.